riscv-osps.adoc

```diff
@@ -0,0 +1,302 @@
+_v1_, if translations exist.  One of the following must hold:
+
+. _v0_ is an impermissible virtual address; the access raises
+  a page-fault exception with trap value _v0_.
+
+. _v0_ is a permissible virtual address; _v1_ lies in a different,
+  impermissible page.  The access raises a page-fault exception with
+  a trap value equal to the base virtual address of the page 
+  containing _v1_.  Alternatively, if the same access to physical-address
+  pair *(* _p0_, _p0_ + _w_ - 1 *)* would have caused an access exception,
+  the implementation may raise that exception instead.  This design
+  simplifies the emulation of misaligned accesses in more-privileged
+  software.
+
+. _v0_ and _v1_ are both permissible virtual addresses.  The access
+  proceeds according to the misaligned physical-memory access rules
+  above, noting that _v0_ and _v1_ may lie in different physical memory
+  regions, despite their virtual contiguity.
+
+##### Memory Maps
+
+#### Required Hardware Components
+##### Interrupt Controller
+##### MMU
+##### DMA
+##### Memory-mapped I/O
+##### IOMMU (for virtualization)
+##### Power Management
+##### CPU Frequency Control
+##### Thermal Management
+
+#### Firmware Requirements
+##### SMBIOS/DMI
+##### IPMI
+##### UEFI
+##### ACPI
+##### Updating Firmware
+
+#### Boot Sequence Requirements
+
+#### Boot Protocol Requirements
+
+
+### Appliance Computing
+This use case sets out the requirements for _appliance computing_.
+For platforms of this type, it is expected that the user will *NOT*
+be able to add or remove cards or devices, with the possible exception
+of adding memory devices such as SD cards.  The hardware in such a
+system is essentially fixed for the life of the device.  Primary
+examples are devices such as tablets, smart phones, automobile systems,
+or even some refrigerators.
+
+The operating system and applications on such appliances may change
+over time.  However, the owner of the device will not be able to
+arbitrarily change the operating system or firmware installed (unless
+they are _really_ clever); in general, this would be fairly strictly
+controlled by the device manufacturer.
+
+Operating systems for this use case could include members of the Linux
+families (Fedora, Debian, Red Hat, SUSE, and similar).  When a platform
+is compliant with the requirements of this use case, it should be
+possible to install and maintain the Linux distributions, but modification
+or specialization of the OS or firmware may be required.
+
+It should also be possible to use any of the other Unix-based operating
+systems and distributions such as those in the BSD family, as well as
+commercial operating systems such as Microsoft Windows.  Again, some
+modification or specialization may be required.
+
+Real-Time Operating Systems (RTOSs) will play a role here as well,
+depending on how the platform is being used.  Whether or not there
+is a standard amenable to all RTOSs is TBD.
```

Comment:
This seems like a big overlap in devices.

RTOSs generally run on small embedded CPUs. These would for example not include an MMU. While phones and tablets are full blown systems that are much more similar to a laptop. Modern phones include vitalisation and IOMMUs for example.

I feel like the `Appliance Computing` section should just be merged into the general computing section. A phone really is just a general computer.

Reply:
Perhaps the general computing section should be written first targeting servers, desktops, and laptops? With that written it might be clear whether it would also work for phones/tablets (yay less work!) or perhaps there will be requirements that don't make sense for them (in which case a "appliance computing" use case is needed)

Reply:
I don't see why a phone or tablet isn't just a general purpose computer. How is a phone different to a raspberry Pi?

I more see that general purpose is everything from a server to a phone. If a user knows what the OS is, then it is general purpose.
With non general purpose being small specialised hardware, maybe running Linux or an embedded OS, but hidden to a user (like a car or fridge).